lib/ddtrace/propagation/distributed_headers.rb

```diff
@@ -53,7 +62,8 @@ def header(name)
 
     def id(header)
       value = header(header).to_i
-      return if value.zero? || value >= Span::MAX_ID
+      # Zero or greater than max allowed value of 2**64
+      return if value.zero? || value > MAX_ID
```

Comment:
I had a minor point: shouldn't `MAX_ID` be a property of the `Span`, not a distributed tracing property? (We're now using `Ext::Distributed::MAX_ID` instead of `Span::MAX_ID`, despite both existing.)

However, I'm not sure I'm understanding this correctly. if we get a value greater than `Span::MAX_ID` but less than `Distributed::MAX_ID`, what's the behavior? Is it coercing the ID into a range that is less than `Span::MAX_ID`?

Just trying to reconcile why we have two `MAX_ID` definitions.

Reply:
The idea is that we will only generate IDs of uint63 (`Span::MAX_ID`), but we allow parsing uint64 (`DistributedTracing::MAX_ID`) IDs for distributed tracing.

Most of our other tracers are doing this as well.

Happy to use a different name for `DistributedTracing::MAX_ID` or moving it somewhere else, open to suggestions.

Reply:
I think we should move this constant to `Span` and rename it something like `MAX_EXTERNAL_ID`, so as to signify that it originates elsewhere. We should also probably add a comment explaining what each max ID constant is, and the difference between the two.

Reply:
I updated the name, that suggestion makes sense :+1: